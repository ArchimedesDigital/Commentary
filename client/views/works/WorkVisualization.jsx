import RaisedButton from 'material-ui/RaisedButton';


// Needed for onTouchTap
// Check this repo:
// https://github.com/zilverline/react-tap-event-plugin
injectTapEventPlugin();

WorkVisualization = React.createClass({

  getChildContext() {
    return { muiTheme: getMuiTheme(baseTheme) };
  },

  childContextTypes: {
    muiTheme: React.PropTypes.object.isRequired
  },

  propTypes: {
    work: React.PropTypes.object.isRequired
  },


  render() {
    let work = this.props.work;
    let work_url = "/commentary/?q=work." + work.slug ;

     return (
       <div className="work-teaser">
         <div class="commentary-text ${work.slug}">

            <a href="/commentary"  >
                <h3 class="text-title">${work.title}</h3>
            </a>

            <hr class="text-divider">

            <div class="text-meta">
            </div>


            <div class="text-subworks " >

                <div class="text-subwork">
                </div>

            </div>



        </div>


        </div>
      );
    }

});
