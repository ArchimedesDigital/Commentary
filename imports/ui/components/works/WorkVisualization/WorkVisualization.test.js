import React from 'react';
import renderer from 'react-test-renderer';

// component:
import WorkVisualization from './WorkVisualization';

describe('WorkVisualization', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(
				<WorkVisualization
					work={{
						title: 'Iliad',
						slug: 'iliad',
						subworks: [{
							title: '1',
							slug: '1',
							n: 1,
						}],
					}}
				/>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
