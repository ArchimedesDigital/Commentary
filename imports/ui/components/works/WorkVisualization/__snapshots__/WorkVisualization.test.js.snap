// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkVisualization renders correctly 1`] = `
<div
  className="work-teaser work-teaser--iliad"
>
  <div
    className="commentary-text"
  >
    <a
      href="/commentary/?q=work.iliad"
    >
      <h3
        className="text-title"
      >
        Iliad
      </h3>
    </a>
    <hr
      className="text-divider"
    />
    <div
      className="text-meta"
    />
    <div
      className="text-subworks text-subworks-visualization-iliad"
    />
  </div>
</div>
`;
